src/Ledger/Ratify.lagda

```diff
@@ -108,15 +111,38 @@ restrictedDists coins rank dists = dists -- record dists { drepStakeDistr = rest
         restrict : Credential ↛ Coin → Credential ↛ Coin
         restrict dist = topNDRepDist rank dist ∪ᵐˡ mostStakeDRepDist dist coins
 
-votedHashes : Vote → ((GovRole × Credential) ↛ Vote) → GovRole → ℙ Credential
+CCData : Set
+CCData = Maybe (KeyHash ↛ Epoch × R.ℚ)
+
+module _ (ce : Epoch) (ccHotKeys : KeyHash ↛ Maybe KeyHash)
+         (cc : CCData) (votes : (GovRole × Credential) ↛ Vote) where
+
+  actualCCVote : KeyHash → Epoch → Vote
+  actualCCVote kh e = case ⌊ ce ≤ᵉ? e ⌋ ,′ lookupᵐ? ccHotKeys kh ⦃ _ ∈? _ ⦄ of λ where
+    (false , just (just hk)) → maybe′ id Vote.no $ lookupᵐ? votes (CC , (inj₁ hk)) ⦃ _ ∈? _ ⦄
```

Comment:
```suggestion
    (true , just (just hk)) → maybe′ id Vote.no $ lookupᵐ? votes (CC , (inj₁ hk)) ⦃ _ ∈? _ ⦄
```
If `ce` is the current epoch and `e` is the end of term, then `ce` should be less than `e` for the vote to be valid.